### PR TITLE
tests: fpu_sharing fix print size issue

### PR DESCRIPTION
**Pull request by hakehuang**
As FPU enalbed the printf code size is changed, so we need add below fix
#30350c91c1ba41033a1c6d1c5b040d1f987c4daf
https://github.com/zephyrproject-rtos/zephyr/pull/30096/commits/30350c91c1ba41033a1c6d1c5b040d1f987c4daf

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

File: tests/kernel/fpu_sharing/generic/prj.conf [hunk 1]
```diff
@@ -3,3 +3,4 @@ CONFIG_FPU=y
 CONFIG_FPU_SHARING=y
 CONFIG_STDOUT_CONSOLE=y
 CONFIG_CBPRINTF_FP_SUPPORT=y
+CONFIG_CBPRINTF_NANO=y
```
Comment (pabigot) [inline tests/kernel/fpu_sharing/generic/prj.conf:6]: This is wrong: the test requires floating point support to display the value of PI.  The nano formatter doesn't support floating point, so with this change it prints:

    Pi calculation OK after 50 (high) + 1 (low) tests (computed %.6lf)

**Review by pabigot — CHANGES_REQUESTED** [review]

File: tests/kernel/fpu_sharing/generic/prj.conf (hunk 1 above)
Comment (hakehuang) [reply tests/kernel/fpu_sharing/generic/prj.conf]: Thanks @pabigot , so the solution shall be increase the main stack size.

**Review by pabigot — APPROVED** [review]
For the record, increasing stack size from 512 by 4 bytes works for float_disable, and by 56 bytes for fpu_sharing.  But the current increase should give some breathing room.

**Comment by hakehuang** [comment]
> For the record, increasing stack size from 512 by 4 bytes works for float_disable, and by 56 bytes for fpu_sharing. But the current increase should give some breathing room.

@pabigot so you suggest to add 64 bytes , instead of add 512 bytes?

**Comment by pabigot** [comment]
> @pabigot so you suggest to add 64 bytes , instead of add 512 bytes?

I don't have an opinion.  512 bytes means this shouldn't have to be fixed for stack overflow for quite a while.  When I added the feature originally, I increased stacks by the minimum necessary to make it pass, to provide evidence that the change did not have a large impact on required stack size, specifically for NANO.

512 is fine with me, I just wanted to verify there wasn't something really surprising going on.

**Comment by hakehuang** [comment]
@nashif  can you help to review again?

**Review by nashif — APPROVED** [review]

**Pull request merged**